nix/nix.go

```diff
@@ -360,7 +361,13 @@ func Push(ctx *ssh.SSHContext, host Host, paths ...string) (err error) {
 		keyArg = "?ssh-key=" + ctx.IdentityFile
 	}
 	if ctx.SkipHostKeyCheck {
-		env = append(env, fmt.Sprintf("NIX_SSHOPTS=%s", "-o StrictHostkeyChecking=No -o UserKnownHostsFile=/dev/null"))
+  		sshOpts = append(sshOpts, fmt.Sprintf("%s", "-o StrictHostkeyChecking=No -o UserKnownHostsFile=/dev/null"))
+	}
+	if ctx.ConfigFile != "" {
+  		sshOpts = append(sshOpts, fmt.Sprintf("-F %s", ctx.ConfigFile))
+	}
+	if len(sshOpts) > 0 {
+  		env = append(env, fmt.Sprintf("NIX_SSHOPTS=%s", sshOpts))
```

Comment:
This is actually not working - it adds an array to the string so you end up with `[ ... ]`. The following works though:
```suggestion
  		env = append(env, fmt.Sprintf("NIX_SSHOPTS=%s", strings.Join(sshOpts, " ")))	
```

Reply:
Thanks :)